lib/spack/spack/compilers/clang.py

```diff
@@ -83,10 +83,7 @@ def is_apple(self):
     @property
     def openmp_flag(self):
         if self.is_apple:
-            raise UnsupportedCompilerFlag(self,
-                                          "OpenMP",
-                                          "openmp_flag",
-                                          "Xcode {0}".format(self.version))
+            return '-Xpreprocessor -fopenmp'
```

Comment:
According to https://iscinumpy.gitlab.io/post/omp-on-high-sierra/, Apple Clang can actually build with OpenMP, you just need to install separate OpenMP libraries. I'm not sure if we want to make this change or not, so if it's controversial, I can remove it.

Reply:
@svenevs You use Apple Clang, right? Do you have any opinions on this change?

Reply:
I'd say it's relatively controversial.  The extra step is usually not taken to install it, so you may be putting things in a situation where the specified flags are not valid.  That would result in compilation failures that may be kind of intimidating for people where `-Xpreprocessor -fopenmp` will fail because they haven't installed it.

Is it allowed to try-compile something (just grab some dummy parallel for loop to test)?  If they have it, it's better to use it than not.

It's worth mentioning that I rarely use OpenMP, in part because support is not good enough for my needs -- AppleClang being the biggest problem.  So my opinion should be taken with a grain of salt, because I'll use full LLVM over AppleClang if I'm on Apple and actually need OpenMP.

Reply:
In that case I think I'll move this change to a separate PR so I can merge the sklearn changes.

Reply:
Sounds good. FYI try-compile may not be enough, it would need to be a try-run or at least try-link I believe. It would probably compile but fail to link if they don't have the libs installed.

It's also worth noting that AppleClang not having full OpenMP has been a problem for years, so I wouldn't personally want to spend a whole lot of time on it ;)

Reply:
@chuckatkins I was just looking at the source code of `Modules/FindOpenMP.cmake` that comes bundled with CMake and it looks like they use `-Xclang` instead of `-Xpreprocessor`. However, https://iscinumpy.gitlab.io/post/omp-on-high-sierra/ and https://scikit-learn.org/dev/developers/advanced_installation.html#macos-compilers-from-homebrew suggest `-Xpreprocessor`. Do you happen to know what the difference is and which Spack should default to? If not, maybe you can ping someone who would know. https://gitlab.kitware.com/cmake/cmake/issues/18520 and https://gitlab.kitware.com/cmake/cmake/merge_requests/1812 also look relevant.